contracts/SquadController.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity >=0.6.0 <0.7.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ERC20Managed.sol";
+import "./ContinuousTokenFactory.sol";
+import "./Curve.sol";
+
+contract SquadController is Ownable, ContinuousTokenFactory {
+    using SafeMath for uint256;
+    using SafeMath for uint16;
+
+    uint16 public networkFee; // in basis points
+
+    address public treasury;
+    mapping(address => uint256) public acounts;
+    uint256 accountsTotal;
+
+    struct Contribution {
+        address beneficiary;
+        uint16 fee; // in basis points
+        uint256 purchasePrice;
+        string contributionURI;
+    }
+    mapping(bytes32 => Contribution) public contributions;
+
+    constructor (
+                 address _reserveToken,
+                 uint16 _networkFee,
+                 address _treasury
+                 ) public ContinuousTokenFactory(_reserveToken) {
+        require(_treasury != address(0), "SquadController: zero treasury address");
+        require(_networkFee <= 10000, "SquadController: network fee > 100%");
+        networkFee = _networkFee;
+        treasury = _treasury;
+    }
+
+    event NewContribution(
+                          address contributor,
+                          bytes32 indexed id,
+                          address indexed beneficiary,
+                          uint16 fee,
+                          uint256 purchasePrice,
+                          address continuousToken,
```

Comment:
This is the address of the ERC20Managed token, right? Is it confusing to call that the continuousToken address?